zipline/data/history_loader.py

```diff
@@ -573,8 +573,9 @@ def _frequency(self):
     @lazyval
     def _calendar(self):
         mm = self.trading_calendar.all_minutes
-        return mm[mm.slice_indexer(start=self._reader.first_trading_day,
-                                   end=self._reader.last_available_dt)]
+        start = mm.searchsorted(self._reader.first_trading_day)
```

Comment:
We've written this algorithm in a whole bunch of places, and every time I do it I have to think nervously about the boundary conditions.  Should we just put a `slice_dates` function in `zipline.utils.pands_utils` that does this?